system-parachains/collectives/collectives-polkadot/src/ambassador/mod.rs

```diff
@@ -0,0 +1,255 @@
+// Copyright (C) 2022 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! The Ambassador Program.
+//!
+//! The module defines the following on-chain functionality of the Ambassador Program:
+//!
+//! - Managed set of program members, where every member has a [rank](ranks)
+//! (via [AmbassadorCollective](pallet_ranked_collective)).
+//! - Referendum functionality for the program members to propose, vote on, and execute
+//! proposals on behalf of the members of a certain [rank](Origin)
+//! (via [AmbassadorReferenda](pallet_referenda)).
+//! - Managed content (charter, announcements) (via [pallet_collective_content]).
+//! - Promotion and demotion periods, register of members' activity, and rank based salaries
+//! (via [AmbassadorCore](pallet_core_fellowship)).
+//! - Members' salaries (via [AmbassadorSalary](pallet_salary), requiring a member to be
+//! imported or inducted into [AmbassadorCore](pallet_core_fellowship)).
+
+pub mod origins;
+mod tracks;
+
+use super::*;
+use crate::xcm_config::{DotAssetHub, FellowshipAdminBodyId};
+use frame_support::traits::{EitherOf, MapSuccess, TryMapSuccess};
+pub use origins::pallet_origins as pallet_ambassador_origins;
+use origins::pallet_origins::{
+	EnsureAmbassadorsVoice, EnsureAmbassadorsVoiceFrom, EnsureHeadAmbassadorsVoice, Origin,
+};
+use parachains_common::polkadot::account;
+use sp_core::ConstU128;
+use sp_runtime::traits::{CheckedReduceBy, ConstU16, ConvertToValue, Replace};
+use xcm::prelude::*;
+use xcm_builder::{AliasesIntoAccountId32, PayOverXcm};
+
+/// The Ambassador Program's member ranks.
+pub mod ranks {
+	use pallet_ranked_collective::Rank;
+
+	#[allow(dead_code)]
+	pub const CANDIDATE: Rank = 0;
+	pub const AMBASSADOR_TIER_1: Rank = 1;
+	pub const AMBASSADOR_TIER_2: Rank = 2;
+	pub const SENIOR_AMBASSADOR_TIER_3: Rank = 3;
+	pub const SENIOR_AMBASSADOR_TIER_4: Rank = 4;
+	pub const HEAD_AMBASSADOR_TIER_5: Rank = 5;
+	pub const HEAD_AMBASSADOR_TIER_6: Rank = 6;
+	pub const HEAD_AMBASSADOR_TIER_7: Rank = 7;
+	pub const MASTER_AMBASSADOR_TIER_8: Rank = 8;
+	pub const MASTER_AMBASSADOR_TIER_9: Rank = 9;
+}
+
+impl pallet_ambassador_origins::Config for Runtime {}
+
+pub type AmbassadorCollectiveInstance = pallet_ranked_collective::Instance2;
+
+/// Demotion is by any of:
+/// - Root can demote arbitrarily.
+/// - the FellowshipAdmin origin (i.e. token holder referendum);
```

Comment:
This looks like it's from the Tech Fellowship, not token holders?

Reply:
it corresponds to the same track in public referendum. I think we can just use the same instead introducing new or setting up to root

Reply:
Ah I see, yeah that's OK then.